cellengine/utils/api_client/APIClient.py

```diff
@@ -119,6 +131,78 @@ def update_entity(self, experiment_id, _id, entity_type, body) -> dict:
             json=body,
         )
 
+    def _get_path(self, entity: CE) -> str:
+        fullpath = f"{self.base_url}/{entity.path}"
+        return fullpath
+
+    # fmt: off
+    # temporary fix for https://github.com/psf/black/issues/1797
+    @overload
+    def create(self, entity: Attachment, **kwargs) -> Attachment: ...
+    @overload
+    def create(self, entity: Compensation, **kwargs) -> Compensation: ...
+    @overload
+    def create(self, entity: Experiment, **kwargs) -> Experiment: ...
+    @overload
+    def create(self, entity: FcsFile, **kwargs) -> FcsFile: ...
+    @overload
+    def create(self, entity: Gate, **kwargs) -> Gate: ...
+    @overload
+    def create(self, entity: Population, **kwargs) -> Population: ...
+    @overload
+    def create(self, entity: ScaleSet, **kwargs) -> ScaleSet: ...
+    # fmt: on
+
+    def create(
+        self, entity, **kwargs
+    ) -> Union[
+        Attachment,
+        Compensation,
+        Experiment,
+        FcsFile,
+        Gate,
+        Population,
+        ScaleSet,
+    ]:
+        """Create a local entity on CellEngine."""
+
+        def unstructure_and_clean(entity, **kwargs):
+            (cls, path, body) = (
+                entity.__class__,
+                entity.path,
+                converter.unstructure(entity),
+            )
+            if body["_id"] == "None" or body["_id"] is None:
+                del body["_id"]  # https://github.com/primitybio/cellengine/issues/5800
+            return (cls, path, body)
+
+        def post_and_restructure(cls, path, body, **kwargs):
+            res = self._post(f"{self.base_url}/{path}", json=body, params=kwargs)
+            return converter.structure(res, cls)
+
+        if isinstance(entity, list):
+            body = [unstructure_and_clean(thing) for thing in entity]
+            things = defaultdict(list)
```

Comment:
I don't think this works. `list` is an argument to the function, but here you're intending to use the `list` class.

(Needs tests!)

Reply:
This is how you're supposed to create a `defaultdict`, in this case where a non-present key will return a default `[]`

Reply:
This is tested in `test_gates.py`, but actually only in the `ge/cattrs-gates` branch (since this is only really relevant to gates). I can test that this doesn't break anything in this branch, but it actually isn't really relevant until the changes in `ge/cattrs-gates`

Reply:
I just pushed a change that uses `create_many` at a file-level; I'm not sure it's necessarily better, but it also works :smile:

Reply:
> This is how you're supposed to create a `defaultdict`, in this case where a non-present key will return a default `[]`

I'm confused...

```py
>>> from collections import defaultdict
>>> x = [1,2]
>>> defaultdict(list)
defaultdict(<class 'list'>, {})
>>> defaultdict(x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: first argument must be callable or None
```

Moot question now though since that code is gone.

Reply:
`defaultdict` wants a callable, not an instantiated class; this is because it contains a `__missing__()` method, which is called if you request a key that doesn't exist. So `defaultdict(list)['notAKey']` will return `list()`

https://docs.python.org/3/library/collections.html